pallets/gear/src/tests.rs

```diff
@@ -9502,3 +9508,86 @@ fn oom_handler_works() {
         );
     });
 }
+
+#[test]
+fn alloc_charge_error() {
+    const WAT: &str = r#"
+(module
+    (import "env" "memory" (memory 1))
+    (import "env" "alloc" (func $alloc (param i32) (result i32)))
+    (export "init" (func $init))
+    (func $init
+        i32.const 0xff
+        call $alloc
+        drop
```

Comment:
I'd add clarification comments on why we get `GasLimitExceeded`. Pretty same as you did with the test below